src/bitbucket-pr.coffee

```diff
@@ -15,243 +11,267 @@
+        @robot.logger.debug "Pull request rejected"
+        @pullRequestDeclined()
 
+      # Merged
+      when @type is 'pullrequest:fulfilled' && 'merged' in ANNOUNCE_OPTIONS
+        @robot.logger.debug "Pull request merged"
+        @pullRequestMerged()
+
+      # Updated
+      when @type is 'pullrequest:updated' && 'updated' in ANNOUNCE_OPTIONS
+        @robot.logger.debug "Pull request updated"
+        @pullRequestUpdated()
+
+      # Approved
+      when @type is 'pullrequest:approved' && 'approve' in ANNOUNCE_OPTIONS
+        @robot.logger.debug "Pull request approved"
+        @pullRequestApproved()
+
+      # Unapproved
+      when @type is 'pullrequest:unapproved' && 'unapprove' in ANNOUNCE_OPTIONS
+        @robot.logger.debug "Pull request unapproved"
+        @pullRequestUnapproved()
+
+  pullRequestCreated: ->
+    "Yo #{@getReviewers()}, #{@actor} just *created* the pull request
+    \"#{@title}\" for `#{@source_branch}` on `#{@repo_name}`.
+    \n#{cached_vars.pr_link}"
+
+  pullRequestCommentCreated: ->
+    "#{@actor} *added a comment* on `#{@repo_name}`:
+    \"#{@resp.comment.content.raw}\"\n#{@resp.comment.links.html.href}"
+
+  pullRequestDeclined: ->
+    @branchAction('declined', 'thwarting the attempted merge of') + "\n#{@pr_link}"
+
+  pullRequestMerged: ->
+    @branchAction('merged', 'joining in sweet harmony')
+
+  pullRequestUpdated: ->
+    @branchAction('updated', 'clarifying why it is necessary to merge') + "\n#{@pr_link}"
+
+  pullRequestApproved: ->
+    "A pull request on `#{@repo_name}` has been approved by #{@actor}
+    \n#{encourageMe()}\n#{@pr_link}"
+
+  pullRequestUnapproved: ->
+    "A pull request on `#{@repo_name}` has been unapproved by #{@actor}\n#{@pr_link}"
+
+class SlackPullRequestEvent extends PullRequestEvent
+  GREEN: '#48CE78'
+  BLUE: '#286EA6'
+  RED: '#E5283E'
+  PURPLE: '#AA82E5'
+  ORANGE: '#F1A56F'
+
+  branchAction: (action_name, color) ->
+    fields = []
+    fields.push
+      title: @title
+      value: @resp.pullrequest.reason
+      short: true
+    fields.push
+      title: "#{@repo_name} (#{@source_branch})"
+      value: @pr_link
+      short: true
+
+    payload =
+      text: "Pull Request #{action_name} by #{@actor}"
+      fallback: "#{@actor} *#{action_name}* pull request \"#{@title}\"."
+      pretext: ''
+      color: color
+      mrkdwn_in: ["text", "title", "fallback", "fields"]
+      fields: fields
+
+  pullRequestCreated: ->
+    reviewers = @getReviewers()
+    content =
+      text: "New Request from #{@actor}"
+      fallback: "Yo #{reviewers}, #{@actor} just *created* the pull request
+                 \"#{@title}\" for `#{@source_branch}` on `#{@repo_name}`."
+      pretext: ''
+      color: @BLUE
+      mrkdwn_in: ["text", "title", "fallback", "fields"]
+      fields: [
+        {
+          title: @title
+          value: "Requesting review from #{reviewers}"
+          short: true
+        }
+        {
+          title: @repo_name
+          value: """
```

Comment:
What do the triple `"` do here? I'm not sensational with Coffeescript, so this could be a clever hack that I'm completely unaware of.